### ISSUE TITLE
"Import into Own Space" not working

### ISSUE THREAD

**Issue by mccalluc** [refinery-platform/refinery-platform#2342]
*(Like the same root cause as #2345.)*

* Specific code commit: 09a2aa2e7e739fb286f370258e899dfe27f5d9c5
* Version of the web browser and OS: FF and Chrome on OSX
* Environment where the error occurred: Production test 

### Steps to reproduce
1. Bring up the preview for an ISAtab, [for example](https://test.stemcellcommons.org/#/preview/07f0edf7-88d2-4e9f-9334-5303aa2685a4)
2. Click "Import into Own Space"

### Observed behavior
* Button turns grey
* 500 from  https://test.stemcellcommons.org/data_set_manager/import/isa-tab-form/ajax/, followed by JS errors:
* Chrome `Cannot read property 'uuid' of undefined at dashboard.js:374`
* FF `Error: this.dataSetDetails is undefined
e.prototype.importDataSet/<@https://refinery-161-testing-static.s3.amazonaws.com/js/dashboard.js:4:25904`
```javascript
    this.dataSetTakeOwnershipService
      .save({
        data_set_uuid: dataSetUuid
      })
      .$promise
      .then(function (data) {
        self.importStatus[dataSetUuid] = { isDataSetReImportSuccess: true };
        self.dashboardDataSetsReloadService.reload(true);
        self.dashboardDataSetPreviewService.preview(data.new_data_set_uuid);
      })
      .catch(function (error) {
        self.importStatus[this.dataSetDetails.uuid] = { isDataSetReImportFail: true };
        self.$log.error(error);
      })
```

### Expected behavior
* You get your own copy of the data set

See also: #2101. Not sure from that description how it would manifest to the user; Possible this is a duplicate?

**Comment by mccalluc** [comment]
Scott does not think this is a dupe. Here's the stack trace... but there are also other errors, earlier in the log:
```
ERROR    django.request:231 handle_uncaught_exception() - Internal Server Error: /data_set_manager/import/isa-tab-form/ajax/
 Traceback (most recent call last):
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/core/handlers/base.py", line 111, in get_response
     response = wrapped_callback(request, *callback_args, **callback_kwargs)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
     return view_func(request, *args, **kwargs)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/views/generic/base.py", line 69, in view
     return self.dispatch(request, *args, **kwargs)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/views/generic/base.py", line 87, in dispatch
     return handler(request, *args, **kwargs)
   File "/srv/refinery-platform/refinery/data_set_manager/views.py", line 194, in get
     download_file(url, temp_file_path)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/celery/app/task.py", line 420, in __call__
     return self.run(*args, **kwargs)
   File "/srv/refinery-platform/refinery/file_store/tasks.py", line 346, in download_file
     "total": remotefilesize
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/celery/app/task.py", line 818, in update_state
     self.backend.store_result(task_id, meta, state)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/celery/backends/base.py", line 259, in store_result
     request=request, **kwargs)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/djcelery/backends/database.py", line 29, in _store_result
     traceback=traceback, children=self.current_task_children(request),
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/djcelery/managers.py", line 42, in _inner
     return fun(*args, **kwargs)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/djcelery/managers.py", line 181, in store_result
     'meta': {'children': children}})
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/djcelery/managers.py", line 87, in update_or_create
     return get_queryset(self).update_or_create(**kwargs)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/djcelery/managers.py", line 70, in update_or_create
     obj, created = self.get_or_create(**kwargs)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/models/query.py", line 424, in get_or_create
     return self._create_object_from_params(lookup, params)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/models/query.py", line 465, in _create_object_from_params
     six.reraise(*exc_info)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/models/query.py", line 457, in _create_object_from_params
     obj.save(force_insert=True, using=self.db)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/models/base.py", line 589, in save
     force_update=force_update, update_fields=update_fields)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/models/base.py", line 617, in save_base
     updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/models/base.py", line 698, in _save_table
     result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/models/base.py", line 731, in _do_insert
     using=using, raw=raw)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/models/manager.py", line 92, in manager_method
     return getattr(self.get_queryset(), name)(*args, **kwargs)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/models/query.py", line 921, in _insert
     return query.get_compiler(using=using).execute_sql(return_id)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 921, in execute_sql
     cursor.execute(sql, params)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
     return self.cursor.execute(sql, params)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
     six.reraise(dj_exc_type, dj_exc_value, traceback)
   File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
     return self.cursor.execute(sql, params)
 IntegrityError: null value in column "task_id" violates not-null constraint
 DETAIL:  Failing row contains (70205, null, PROGRESS, gAJ9cQEoVQdjdXJyZW50cQJNig5VDHBlcmNlbnRfZG9uZXEDVQcxMDAuMDAlVQV0..., 2017-11-08 19:35:38.756372+00, null, f, eJxrYKotZIzgYGBgSM7IzEkpSs0rZIotZC7WAwBWuwcA).
```

**Comment by mccalluc** [comment]
The last point in the stack in our code before it goes to libraries is here: 
```
            import_file.update_state(
                state="PROGRESS",
                meta={
                    "percent_done": "%3.2f%%" % percent_done,
                    "current": localfilesize,
                    "total": remotefilesize
                }
            )
```

What does the null `task_id` mean? Did something return a null when it should have errored out?

**Comment by scottx611x** [comment]
I'm also seeing the following errors, could be related:
```
 2017-11-08 14:58:55 ERROR    django.request:231 handle_uncaught_exception() - Internal Server Error: /data_set_manager/import/isa-tab-form/
Traceback (most recent call last):
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/core/handlers/base.py", line 111, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/views/generic/base.py", line 69, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/views/generic/base.py", line 87, in dispatch
    return handler(request, *args, **kwargs)
  File "/srv/refinery-platform/refinery/data_set_manager/views.py", line 200, in get
    context_instance=context)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/shortcuts.py", line 25, in render_to_response
    return HttpResponse(loader.render_to_string(*args, **kwargs), **httpresponse_kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/template/loader.py", line 170, in render_to_string
    t = get_template(template_name, dirs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/template/loader.py", line 144, in get_template
    template, origin = find_template(template_name, dirs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/template/loader.py", line 136, in find_template
    raise TemplateDoesNotExist(name)
```

**Comment by mccalluc** [comment]
`TemplateDoesNotExist` is in the screenshots for #2101, so I still wonder if there is some overlap.

**Comment by mccalluc** [comment]
Successfully imported "Transcription factor binding during intestinal cell differentiation [TFB]" into own space on beta.

Next want to try for a local reproducer.

**Comment by scottx611x** [comment]
@mccalluc I could reproduce this locally, also #2346 Does not address this issue

**Comment by scottx611x** [comment]
Fixed by #2346
